cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/service/DatasetInstanceService.java

```diff
@@ -179,9 +229,11 @@ private DatasetMeta getFromMds(Id.DatasetInstance instance) throws Exception {
    * @throws DatasetTypeNotFoundException if the dataset type was not found
    * @throws Exception if something went wrong
    */
-  public void create(String namespaceId, String name, DatasetInstanceConfiguration props) throws Exception {
+  void create(String namespaceId, String name, DatasetInstanceConfiguration props) throws Exception {
     // Throws NamespaceNotFoundException if the namespace does not exist
     Id.Namespace namespace = ConversionHelpers.toNamespaceId(namespaceId);
+    Principal principal = SecurityRequestContext.toPrincipal();
```

Comment:
The call may not come from HTTP. Shouldn't we always use the logic above that try to get it from SQC and fall back to UGI?